google-cloud-logging/src/test/java/com/google/cloud/logging/ContextTest.java

```diff
@@ -114,40 +127,78 @@ public void testParsingCloudTraceContext() {
     final String X_CLOUD_TRACE_NO_TRACE = "/SPAN_ID;o=TRACE_TRUE";
     final String X_CLOUD_TRACE_ONLY = TEST_TRACE_ID;
     final String X_CLOUD_TRACE_WITH_SPAN = TEST_TRACE_ID + "/" + TEST_SPAN_ID;
-    final String X_CLOUD_TRACE_FULL = TEST_TRACE_ID + "/" + TEST_SPAN_ID + ";o=TRACE_TRUE";
+    final String X_CLOUD_TRACE_FULL = TEST_TRACE_ID + "/" + TEST_SPAN_ID + ";o=1";
 
     Context.Builder builder = Context.newBuilder();
 
     builder.loadCloudTraceContext(null);
-    assertTraceAndSpan(builder.build(), null, null);
+    assertTraceSpanAndSampled(builder.build(), null, null, false);
     builder.loadCloudTraceContext("");
-    assertTraceAndSpan(builder.build(), null, null);
+    assertTraceSpanAndSampled(builder.build(), null, null, false);
     builder.loadCloudTraceContext(X_CLOUD_TRACE_NO_TRACE);
-    assertTraceAndSpan(builder.build(), null, null);
+    assertTraceSpanAndSampled(builder.build(), null, null, false);
     builder.loadCloudTraceContext(X_CLOUD_TRACE_ONLY);
-    assertTraceAndSpan(builder.build(), TEST_TRACE_ID, null);
+    assertTraceSpanAndSampled(builder.build(), TEST_TRACE_ID, null, false);
     builder.loadCloudTraceContext(X_CLOUD_TRACE_WITH_SPAN);
-    assertTraceAndSpan(builder.build(), TEST_TRACE_ID, TEST_SPAN_ID);
+    assertTraceSpanAndSampled(builder.build(), TEST_TRACE_ID, TEST_SPAN_ID, false);
     builder.loadCloudTraceContext(X_CLOUD_TRACE_FULL);
-    assertTraceAndSpan(builder.build(), TEST_TRACE_ID, TEST_SPAN_ID);
+    assertTraceSpanAndSampled(builder.build(), TEST_TRACE_ID, TEST_SPAN_ID, TEST_TRACE_SAMPLED);
   }
 
   @Test
   public void testParsingW3CTraceParent() {
     final String W3C_TEST_TRACE_ID = "12345678901234567890123456789012";
     final String W3C_TEST_SPAN_ID = "1234567890123456";
-    final String W3C_TRACE_CONTEXT = "00-" + W3C_TEST_TRACE_ID + "-" + W3C_TEST_SPAN_ID + "-00";
+    final String W3C_TEST_TRACE_SAMPLED = "0f";
+    final String W3C_TRACE_CONTEXT =
+        "00-" + W3C_TEST_TRACE_ID + "-" + W3C_TEST_SPAN_ID + "-" + W3C_TEST_TRACE_SAMPLED;
 
     Context.Builder builder = Context.newBuilder();
 
     builder.loadW3CTraceParentContext(null);
-    assertTraceAndSpan(builder.build(), null, null);
+    assertTraceSpanAndSampled(builder.build(), null, null, false);
     builder.loadW3CTraceParentContext(W3C_TRACE_CONTEXT);
-    assertTraceAndSpan(builder.build(), W3C_TEST_TRACE_ID, W3C_TEST_SPAN_ID);
+    assertTraceSpanAndSampled(builder.build(), W3C_TEST_TRACE_ID, W3C_TEST_SPAN_ID, true);
   }
 
-  private void assertTraceAndSpan(Context context, String expectedTraceId, String expectedSpanId) {
+  @Test
+  public void testParsingOpenTelemetryContext() {
```

Comment:
Have you tested the case where you have an OpenTelemetry span and a request with either a Traceparent or XCTC header?

Reply:
I have added some system tests here: https://github.com/googleapis/java-logging/pull/1596/files#diff-116da1fe69768cf63bc5b434e783fc90c20d21f1a437d2ddab7d6b677f56ea8e